compiler/noirc_frontend/src/parser/parser/traits.rs

```diff
@@ -12,34 +17,115 @@ use super::Parser;
 
 impl<'a> Parser<'a> {
     /// Trait = 'trait' identifier Generics ( ':' TraitBounds )? WhereClause TraitBody
+    ///       | 'trait' identifier Generics '=' TraitBounds WhereClause ';'
     pub(crate) fn parse_trait(
         &mut self,
         attributes: Vec<(Attribute, Span)>,
         visibility: ItemVisibility,
         start_span: Span,
-    ) -> NoirTrait {
+    ) -> (NoirTrait, Option<NoirTraitImpl>) {
         let attributes = self.validate_secondary_attributes(attributes);
 
         let Some(name) = self.eat_ident() else {
             self.expected_identifier();
-            return empty_trait(attributes, visibility, self.span_since(start_span));
+            let noir_trait = empty_trait(attributes, visibility, self.span_since(start_span));
+            let no_implicit_impl = None;
+            return (noir_trait, no_implicit_impl);
         };
 
         let generics = self.parse_generics();
-        let bounds = if self.eat_colon() { self.parse_trait_bounds() } else { Vec::new() };
-        let where_clause = self.parse_where_clause();
-        let items = self.parse_trait_body();
 
-        NoirTrait {
+        // Trait aliases:
+        // trait Foo<..> = A + B + E where ..;
+        let (bounds, where_clause, items, is_alias) = if self.eat_assign() {
```

Comment:
Optional: instead of desugaring a trait alias in the parser itself, we could introduce a new AST node for it (maybe `TraitAlias`) and desugar it later on, maybe in the Elaborator or collector. That way:
- parsing is a bit simpler
- desugaring can be tested independently
- LSP or tools working with parsed code (like the formatter) don't need to know about how trait aliases are expanded, knowing that they need to skip the synthetic impl, and can focus on better handling this new node

Reply:
I originally attempted to implement this by desugaring it in the `Elaborator` (based on the `is_alias` flag), but it was fairly complex to deal with post-parsing types.
- Testing this approach seems a bit simpler since we can check ~equality up to spans between the parsed trait impl and the expected desugaring.

I suppose adding an "is_synthetic: bool` to `NoirTraitImpl` would be easy if that helps handling it in the LSP?

Sure, I'll add a formatter test for this 👍

Reply:
@asterite I've added `is_synthetic: bool` to `NoirTraitImpl` and formatter support for trait aliases, with tests for both.